user_sync/config.py

```diff
@@ -84,6 +87,24 @@ def set_options(self, caller_options):
     def get_logging_config(self):
         return self.main_config.get_dict_config('logging', True)
 
+    def validate_limits_config(self):
+        ''' Ensure that we have limits set for max deletions per run and max missing users
+        '''
+        limits_config = self.main_config.get_dict_config('limits', True)
+        if limits_config == None:
+            raise user_sync.error.AssertionException("Limits configuration must be specified.")
+
+        max_deletions_per_run = limits_config.get_int('max_deletions_per_run',True)
+        if not max_deletions_per_run:
+            raise user_sync.error.AssertionException("max_deletions_per_run must be specified.")
+
+        max_missing_users = limits_config.get_int('max_missing_users',True)
+        if not max_missing_users:
```

Comment:
Do we even need to validate_limits_config?   If you use the approach that I described and access the fields as is in get_rule_options, we don't need to push the values into options then into rule options.